containers/podman-desktop
# feat: create informers for kube contexts

Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

- adds a new boolean Preference entry `Get information about clusters declared in Kubeconfig`, to enable / disable the running of informers watching pods/deployments/replicasets for different kube contexts
- if enabled, the kubernetes client singleton manages a state of informers for the different contexts (stops informers of delete contexts, starts new informers for new contexts)
- if enabled, displays in the list of kube contexts if the cluster is reachable, and the number of current pods/deployemnts/replicasets for each kube context

### Screenshot / video of UI



### What issues does this PR fix or reference?

#4562



### How to test this PR?

- Enable the feature in the Preferences
- Switch contexts and check information is updated
- add/remove resources (pods, deployments) from context and check information is updated
-
